### PR TITLE
add libmysqlclient 8.0.24

### PR DESCRIPTION
**Pull request by AndreyMlashkin**
library name and version:  **libmysqlclient/8.0.24**

implements https://github.com/conan-io/conan-center-index/issues/5455

I need some help here - the library is compiled, but can't be linked in test package
---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by AndreyMlashkin** [comment]
It seems, that the library is not packaged:
```
~/.conan/data/libmysqlclient/8.0.17$ find -name "libmysqlclient.a"
./_/_/build/c4df4661a2eec2a5c4be03c1632671566d378dac/archive_output_directory/libmysqlclient.a
./_/_/package/c4df4661a2eec2a5c4be03c1632671566d378dac/lib/libmysqlclient.a

~/.conan/data/libmysqlclient/8.0.24$ find -name "libmysqlclient.a"
./_/_/build/6cda070908f6f9922bbb2d1a681a5d71f4a3ff97/archive_output_directory/libmysqlclient.a
```

**Comment by AndreyMlashkin** [comment]
> Having a quick comparison it looks like there is no need to create a new `8.0.24/conanfile.py`. Please add this version to `all/`

There were many cosmetic changes and none of the old patches can be applied.

here is what chaged, see attachment
[patches_diff.txt](https://github.com/conan-io/conan-center-index/files/6457991/patches_diff.txt)

diff all/patches/ 8.0.24/patches/ > patches_diff.txt

**Comment by Croydon** [comment]
You don't have to apply the old patches to newer libmysqlclient versions. You can specify within `conandata.yml` which patches apply to which version. You can find an example here: https://github.com/conan-io/conan-center-index/blob/ded6a54567baf1cf6c6a06447d44fe261e92ec8a/recipes/gtest/all/conandata.yml#L11

**Comment by AndreyMlashkin** [comment]
I still need some help with library export

File: recipes/libmysqlclient/all/conanfile.py [hunk 1]
```diff
@@ -49,7 +49,10 @@ def _patch_files(self):
         os.rename(sources_cmake, sources_cmake_orig)
         os.rename("CMakeLists.txt", sources_cmake)
         if self.settings.os == "Macos":
-            tools.replace_in_file(os.path.join(self._source_subfolder, "libmysql", "CMakeLists.txt"), "COMMAND $<TARGET_FILE:libmysql_api_test>", "COMMAND DYLD_LIBRARY_PATH=%s $<TARGET_FILE:libmysql_api_test>" % os.path.join(self.build_folder, "library_output_directory"))
+            if self.version <= "8.0.17":
```
Comment (uilianries) [inline recipes/libmysqlclient/all/conanfile.py:52]:
```suggestion
            if tools.Version(self.version) <= "8.0.17":
```

self.version is a string, need to convert to Version class.

**Comment by AndreyMlashkin** [comment]
Seems like mysqlclient library is not packaged on some reason. Need some help here, what's the best way to do so - just to copy it or should I add something to package_info?

**Comment by AndreyMlashkin** [comment]
Could someone help please?
I see that authentication_ldap_sasl_client is in lib folder and is in collect_libs() list, but it is still can't be found

File: recipes/libmysqlclient/all/conanfile.py [hunk 2]
```diff
@@ -31,6 +31,8 @@ def requirements(self):
         if self.options.with_zlib:
             self.requires("zlib/1.2.11")
 
+        self.requires("lz4/1.9.3")
```
Comment (prince-chrismc) [inline recipes/libmysqlclient/all/conanfile.py:34]: Is this version dependant? it was not required before?

File: recipes/libmysqlclient/all/conanfile.py [hunk 3]
```diff
@@ -49,7 +51,8 @@ def _patch_files(self):
         os.rename(sources_cmake, sources_cmake_orig)
         os.rename("CMakeLists.txt", sources_cmake)
```
Comment (prince-chrismc) [inline recipes/libmysqlclient/all/conanfile.py:52]: Just some conventions

```suggestion
        tools.rename(sources_cmake, sources_cmake_orig)
        tools.rename("CMakeLists.txt", sources_cmake)
```

File: recipes/libmysqlclient/all/patches/0009-remove-man-8.0.24.patch [hunk 4]
```diff
@@ -0,0 +1,11 @@
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -1654,7 +1654,7 @@ IF(ENABLE_GCOV)
+   INCLUDE(fastcov)
+ ENDIF()
+ 
+-IF(UNIX)
++IF(FALSE)
```
Comment (prince-chrismc) [inline recipes/libmysqlclient/all/patches/0009-remove-man-8.0.24.patch:8]: Seems like the recipe deletes these files is this required?

File: recipes/libmysqlclient/all/patches/0007-msvc-install-no-pdb-8.0.24.patch [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
++  IF(FALSE)
+     GET_TARGET_PROPERTY(type ${target} TYPE)
+     IF(NOT INSTALL_LOCATION)
+       IF(type MATCHES "STATIC_LIBRARY"
+@@ -110,7 +110,7 @@ FUNCTION(MYSQL_INSTALL_TARGET target_arg)
+   IF(ARG_COMPONENT)
+     SET(COMP COMPONENT ${ARG_COMPONENT})
+   ENDIF()
+-  INSTALL(TARGETS ${target} DESTINATION ${ARG_DESTINATION} ${COMP})
++  INSTALL(TARGETS ${TARGETS} RUNTIME DESTINATION "${CMAKE_INSTALL_BINDIR}" ARCHIVE DESTINATION "${CMAKE_INSTALL_LIBDIR}" LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}" ${COMP})
```
Comment (prince-chrismc) [inline recipes/libmysqlclient/all/patches/0007-msvc-install-no-pdb-8.0.24.patch:17]: These are deleted in the recipe as well, seems a touch extra

**Comment by prince-chrismc** [comment]
Looking at the logs... there are only a few failures

image

> Could someone help please?
> I see that authentication_ldap_sasl_client is in lib folder and is in collect_libs() list, but it is still can't be found

Seems the question is solely in apple-clang 12 

Right away my eye's tripped on 

```txt
libmysqlclient/8.0.24 package(): Packaged 10 '.a' files
libmysqlclient/8.0.24 package(): Packaged 1 '.so' file: authentication_ldap_sasl_client.so
```

Why is this the only one being compiled as shared? And in the profile it's a static build

I see the it's the same problem on more compilers in the older version

**Comment by prince-chrismc** [comment]
@ericLemanissier has #5677 open and passing

**Comment by AndreyMlashkin** [comment]
Ok, I will close this one

**Pull request closed without merging**